power_grid_model_c/power_grid_model/include/power_grid_model/component/fault.hpp

```diff
@@ -125,6 +125,22 @@ class Fault final : public Base {
         return fault_type_;
     }
     FaultPhase get_fault_phase() const {
+        using enum FaultPhase;
+
+        if (fault_phase_ == default_value || fault_phase_ == FaultPhase::nan) {
+            switch (fault_type_) {
+                case FaultType::three_phase:
+                    return abc;
+                case FaultType::single_phase_to_ground:
+                    return a;
+                case FaultType::two_phase:
+                    [[fallthrough]];
+                case FaultType::two_phase_to_ground:
+                    return bc;
+                case FaultType::nan:
```

Comment:
`fault_type` is a mandatory attribute, so it will never has the value `nan` in the class object. So here you need to throw `InvalidEnum`, because it should not happen.

Reply:
it could happen that the user sets it to `nan` in the input data (constructor) and then does a later batch update where he sets the value, right?

Reply:
This function `get_fault_phase` is not called in the constructor right? Only at the moment you call `calc_param`. There you should have a valid `fault_type`.